news/_posts/2020-02-04-the-new-prestashop-test-framework.md

```diff
@@ -0,0 +1,95 @@
+- Selenium itself is not very stable: the test campaign could be run 3 or 4 times in a row on Travis and have different results, with different tests failing. Sometimes it could not find a selector, or [timed out](https://github.com/PrestaShop/PrestaShop/issues/14384) when waiting for something… it was very difficult and time-consuming to make the tests more robust.
+- The campaign was too complex: it was trying to test everything at once. The campaign would test all the functionalities in a page, but also doing some [integration testing](https://devdocs.prestashop.com/1.7/testing/how-to-create-your-own-web-acceptance-tests/#functional-tests) (verifying some functionalities work well together) and some [end-to-end testing](https://devdocs.prestashop.com/1.7/testing/how-to-create-your-own-web-acceptance-tests/#end-to-end-tests-soon) (running through the application like a real user and testing a bunch of functionalities at once). The result was a very complex test suite, very brittle (end-to-end tests are really fragile by design) and a nightmare to maintain and fix.
+
+There were some concerns with the structure of the code of this campaign too. Although it did the job, there was no clear architecture behind it. No separation of concerns, no proper use of inheritance, no distinction between the test logic and the page logic. The lack of structure made it very complicated to find the source of a problem sometimes, and with more than 8000 test cases this was a considerable waste of time.
+
+All these reasons made it clear that it was time to drop the whole thing and start over.
+
+
+## What the team is building now
+
+The QA team decided to work on the three following points.
+
+### Automation tool
+
+It was already decided to drop Selenium, but what would be a good replacement? There are a lot of options there, including SaaS solutions ([Cypress](https://www.cypress.io/), [QAWolf](https://docs.qawolf.com/), etc). After reviewing a few possibilities, the best choice was to go for [Puppeteer](https://github.com/puppeteer/puppeteer).
+
+Puppeteer is a Node library which provides a nice API to control Chrome. It can automate pretty much everything you do manually. It is maintained by Google, which is considered a good indicator of the quality of a project.
+
+
+But more importantly, it’s more stable in headless (which was a killer feature for us !) since it was designed with headless in mind.
```

Comment:
```suggestion
But more importantly, it’s more stable in headless (which was a killer feature for us!) since it was designed with headless in mind.
```